### DISCUSSION TITLE
Bugs / choppy diurnal motion for mobile planetariums

### DISCUSSION THREAD

**Discussion by Starrsplanetarium** [Stellarium/stellarium#3658] [General]
After playing with a few versions of Stellarium I have realised that the choppiness of some versions is much more pronounced when the spheric mirror distortion option is toggled on.
Maybe this is something that the developers can look at for future versions.
Cheers

Gary Starr
Starr's Planetarium

**Comment by ImmersiveTheatres** [comment]
I tested numerous versions on a brand new MacBook in December (with spherical mirror projection), and found most were choppy with diurnal motion. 

Yet a never realised the choppiness is linked to spherical mirror projection.

Hopefully this is a useful clue / detail for the developers?

**Comment by gzotti** [comment]
A problem is that the currently 3 active developers, sitting behind rectangular screens, rarely think of the spherical mirror distortion which has been introduced before they started their contributions, i.e. this is pretty old legacy code that is still expected to "just work as before". When a developer detects loss of performance while adding features, usually we do something against it before even pushing such change. However we cannot test performance impact of each and every change in each and every possible installation, and don't have the hardware zoo out there Stellarium is supposed to run on.  It may help if you could list versions that work better or worse, and add your system specs, so that we _may_ have a chance to track down at least the 3-month period between releases where performance-relevant changes happened. When you say something like 0.16 worked on your older Mac, but 23.3 has issues on an M3, there are just too many variables.

**Reply by Starrsplanetarium** [reply]
After reinstalling Windows I installed the latest Stellarium version and then tried a few previous versions. Sorry, but I didn't note down the versions that I tried.
I have now gone back to V 0.22.2 which works well on my machine.
Windows 10
Intel Core i9
NVIDIA Quatro TRX 3000
32G RAM

Cheers.

**Comment by 10110111** [comment]
> Yet a never realised the choppiness is linked to spherical mirror projection.

This projection just takes additional computational load on the system, adding to time taken per frame. It's not the single reason for choppiness.

You may find `--low-graphics` option in the latest weekly build useful to improve performance at the cost of some degradation of the visuals.
